src/pages/api/products/[id].ts

```diff
@@ -0,0 +1,33 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+
+const productHandler = (req: NextApiRequest, res: NextApiResponse) => {
+  const { method, query: { id } } = req
+
+  // TODO: Handle the function properly
+  const handlePatch = () => {
+    return {
+      message: `This is a PATCH request, it should patch id: ${id}. Currently in progress...`,
+      id: id
+    }
+  }
+
+  // TODO: Handle the function properly
+  const handleDelete = () => {
+    return {
+      message: `This is a DELETE request, it should DELETE id: ${id}. Currently in progress...`,
+      id: id
+    }
+  }
+
+  switch (method) {
+    case 'PATCH':
+      return res.json(handlePatch())
```

Comment:
I am not quite familiar with this approach ( `res.json(handle...)` ), Do people normally use the word `handle` to name their function that **returns object**?. If so please provide me ref(s). 
Personally, It think it looks cleaner if we could possibly make something like this below. What do you think?

```
  switch (method) {
    case 'POST': 
      handlePost()
      break;
    case 'PATCH': 
      handlePatch()
      break;
    default: ...
```

Reply:
Other developers preferences: https://stackoverflow.com/questions/6114210/is-returning-out-of-a-switch-statement-considered-a-better-practice-than-using-b
Facts (See last paragraph, last sentence): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#description